model_compression_toolkit/qat/common/qat_get_quantizer_config.py

```diff
@@ -0,0 +1,106 @@
+    """
+    final_cfg = n.final_activation_quantization_cfg
+    return TrainableQuantizerActivationConfig(final_cfg.activation_quantization_method,
+                                              final_cfg.activation_n_bits,
+                                              final_cfg.activation_quantization_params,
+                                              final_cfg.enable_activation_quantization,
+                                              final_cfg.min_threshold,
+                                              activation_quantization_candidates)
+
+
+def get_trainable_quantizer_quantization_candidates(n: common.BaseNode):
+    """
+    Returns quantization configuration candidates for activation and weights trainable quantizer
+
+    Args:
+        n: BaseNode
+
+    Returns:
+         weights_quantization_candidates - A list of configuration candidates for weights
+         activation_quantization_candidates - A list of configuration candidates for activation
+    """
+
+    # all candidates must have the same weights quantization method
+    weights_quantization_method = len(set([cfg.weights_quantization_cfg.weights_quantization_method for cfg in n.candidates_quantization_cfg]))
+    if weights_quantization_method > 1:
+        common.Logger.error(f'Unsupported candidates_quantization_cfg with different weights quantization methods')
+
+    # all candidates must have the same activation quantization method
+    activation_quantization_method = len(set([cfg.activation_quantization_cfg.activation_quantization_method for cfg in n.candidates_quantization_cfg]))
+    if activation_quantization_method > 1:
+        common.Logger.error(f'Unsupported candidates_quantization_cfg with different activation quantization methods')
+
+    # get unique lists of candidates
+    unique_weights_candidates = n.get_unique_weights_candidates()
+    unique_activation_candidates = n.get_unique_activation_candidates()
+
+    # verify all the combinations of weights_n_bits and activation_n_bits are allowed
+    if len(n.candidates_quantization_cfg) != len(unique_weights_candidates) * len(unique_activation_candidates):
+        common.Logger.error(f'Unsupported candidates_quantization_cfg for a trainable quantizer,'
+                            f'it must contain all the combinations of (weights_n_bits X activations_n_bits)')
+
+    # generate list of weights quantizer candidates
+    weights_cfg_candidates = [TrainableQuantizerCandidateConfig(
+        cfg.weights_quantization_cfg.weights_n_bits,
+        cfg.weights_quantization_cfg.weights_quantization_params) for cfg in unique_weights_candidates]
+
+    # generate list of activation quantizer candidates
+    activation_cfg_candidates = [TrainableQuantizerCandidateConfig(
+        cfg.activation_quantization_cfg.activation_n_bits,
+        cfg.activation_quantization_cfg.activation_quantization_params) for cfg in unique_activation_candidates]
+
+    return weights_cfg_candidates, activation_cfg_candidates
```

Comment:
Maybe this could be split into two functions. One for weights and one for activations?
And the assertion of non-missing configuration (the cartesian product) should be outside after calling them?